conda/models/match_spec.py

```diff
@@ -33,6 +33,17 @@
 
 class MatchSpecType(type):
     def __call__(cls, spec_arg=None, **kwargs):
+        try:
+            return cls._call(spec_arg, **kwargs)
+        except InvalidSpec as e:
+            msg = ""
+            if spec_arg:
+                msg += f"{spec_arg}"
+            if kwargs:
+                msg += " " + ", ".join(f"{k}={v}" for k, v in kwargs.items())
+            raise InvalidMatchSpec(msg, details=e) from e
+
+    def _call(cls, spec_arg=None, **kwargs):
```

Comment:
This looks off due to the name of the argument, it takes a class object and is called from the __call__ method, I think this should be a classmethod and not an instance method to not be handed down to MatchSpec instances?

Reply:
I'm borrowing the signature from `__call__` above. I'm not sure what gets passed to the method here, since we are in the Metaclass land now. Should `__call__` also be a classmethod by that definition?

Reply:
Reading up on this a little more, you're right, the __call__ method of type subclasses get passed the class and not instances and it's a misleading case of this meta-programming pattern that the _call method looks like taking an instance. This is bound to confuse people (as it did me), so I would suggest to simply not call a separate method with a confusing name `_call` and just inline the try/except block. That should suffice to handle this and not hurt us long-term.

Reply:
Done!